indico/core/signals/menu.py

```diff
@@ -0,0 +1,32 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+
+from blinker import Namespace
+
+_signals = Namespace()
+
+
+items = _signals.signal('menu-items', """
+Expected to return `(menu_item_name, SideMenuItem)` tuples to be added to
```

Comment:
If you used `named_object_from_signal` instead of `values_from_signal` and added a `name` attribute to the menu sections/items you could simply return/yield items instead of tuples. Also, you'd get a check to ensure unique names for free ;)